executor/executor.go

```diff
@@ -571,6 +594,67 @@ func (e *SelectionExec) Next(goCtx goctx.Context) (Row, error) {
 	}
 }
 
+// NextChunk implements the Executor NextChunk interface.
+func (e *SelectionExec) NextChunk(chk *chunk.Chunk) error {
+	chk.Reset()
+
+	if !e.batched {
+		return errors.Trace(e.unBatchedNextChunk(chk))
+	}
+
+	for {
+		for ; e.inputRow != e.childrenResults[0].End(); e.inputRow = e.inputRow.Next() {
+			if !e.selected[e.inputRow.Idx()] {
+				continue
+			}
+			chk.AppendRow(0, e.inputRow)
+			if chk.NumRows() == e.ctx.GetSessionVars().MaxChunkSize {
```

Comment:
1. move this check to line 610 thus line 612 can be removed.
2. chk.reachMaxSize()